qiskit_ibm_provider/utils/json_decoder.py

```diff
@@ -69,6 +83,152 @@ def properties_from_server_data(properties: Dict) -> BackendProperties:
+    qiskit_control_flow_mapping = {
+        "if_else": IfElseOp,
+        "while_loop": WhileLoopOp,
+        "for_loop": ForLoopOp,
+    }
+
+    # Create instruction property placeholder from backend configuration
+    supported_instructions = set(getattr(configuration, "supported_instructions", []))
+    basis_gates = set(getattr(configuration, "basis_gates", []))
+    gate_configs = {gate.name: gate for gate in configuration.gates}
+    inst_name_map = {}  # type: Dict[str, Instruction]
+    prop_name_map = {}  # type: Dict[str, Dict[Tuple[int, ...], InstructionProperties]]
+    all_instructions = set.union(supported_instructions, basis_gates, set(required))
+
+    # Create name to Qiskit instruction object repr mapping
+    for name in all_instructions:
+        if name in qiskit_control_flow_mapping:
+            continue
+        if name in qiskit_inst_mapping:
+            inst_name_map[name] = qiskit_inst_mapping[name]
+        elif name in gate_configs:
+            this_config = gate_configs[name]
+            params = list(map(Parameter, getattr(this_config, "parameters", [])))
+            coupling_map = getattr(this_config, "coupling_map", [])
+            inst_name_map[name] = Gate(
+                name=name,
+                num_qubits=len(coupling_map[0]) if coupling_map else 0,
+                params=params,
+            )
+        else:
+            warnings.warn(
+                f"Definition of instruction {name} is not found in Qiskit and "
+                "no gate configuration is provided by the backend. "
+                "This instruction is not created in Target.",
+                UserWarning,
+            )
+            all_instructions.remove(name)
+            continue
+
+    # Create placeholder for the instruction properties
+    for name, spec in gate_configs.items():
+        if hasattr(spec, "coupling_map"):
+            coupling_map = spec.coupling_map
+            prop_name_map[name] = dict.fromkeys(map(tuple, coupling_map))
+        else:
+            prop_name_map[name] = None
+    if "delay" not in prop_name_map:
+        # Case for real IBM backend. They don't have delay in gate configuration.
+        prop_name_map["delay"] = {(q,): None for q in range(configuration.n_qubits)}
+
+    # Populate instruction properties
+    if properties:
+        in_data["qubit_properties"] = list(
+            map(_decode_qubit_property, properties["qubits"])
+        )
+        for gate_spec in map(GateSchema.from_dict, properties["gates"]):
+            inst_prop = _decode_instruction_property(gate_spec)
+            qubits = tuple(gate_spec.qubits)
+            name = gate_spec.gate
+            if name not in all_instructions:
+                # New instruction is found in property. Likely an edge case.
+                new_instruction = Gate(
+                    name=name,
+                    num_qubits=len(qubits),
+                    params=[],
+                )
```

Comment:
Shouldn't we look this up in `qiskit_inst_mapping` first instead of assigning it an opaque gate?